lib/upgrade/etcd/conversionv1v3/bgppeer.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"github.com/projectcalico/libcalico-go/lib/scope"
+)
+
+// BGPPeer implements the Converter interface.
+type BGPPeer struct{}
+
+// APIV1ToBackendV1 converts v1 BGPPeer API to v1 BGPPeer KVPair.
+func (bp BGPPeer) APIV1ToBackendV1(rIn unversioned.Resource) (*model.KVPair, error) {
+	ap := rIn.(apiv1.BGPPeer)
```

Comment:
I know I haven't done this in the IPPool example, but we should make sure the interface assertion works else it could panic. So like:
```go
ap, ok := rIn.(apiv1.BGPPeer)
if !ok {
 log.Fatalf(...)
}
```

Reply:
Sounds good, I had thought about that but didn't add it in since it wasn't in the example.

Reply:
Did you specifically want a log.Fatal and not return an error?

Reply:
I meant return error, sorry